crates/nostr-mls/src/extension.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) 2024-2025 Jeff Gardner
+// Copyright (c) 2025 Rust Nostr Developers
+// Distributed under the MIT software license
+
+//! Nostr Group Extension functionality for MLS Group Context.
+//! This is a required extension for Nostr Groups as per NIP-104.
+
+use std::collections::BTreeSet;
+use std::str;
+
+use nostr::secp256k1::rand::rngs::OsRng;
+use nostr::secp256k1::rand::Rng;
+use nostr::util::hex;
+use nostr::{PublicKey, RelayUrl};
+use openmls::extensions::{Extension, ExtensionType};
+use openmls::group::{GroupContext, MlsGroup};
+use tls_codec::{
+    DeserializeBytes, TlsDeserialize, TlsDeserializeBytes, TlsSerialize, TlsSerializeBytes, TlsSize,
+};
+
+use crate::constant::NOSTR_GROUP_DATA_EXTENSION_TYPE;
+use crate::error::Error;
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    TlsSerialize,
+    TlsDeserialize,
+    TlsDeserializeBytes,
+    TlsSerializeBytes,
+    TlsSize,
+)]
+pub(crate) struct RawNostrGroupDataExtension {
+    pub nostr_group_id: [u8; 32],
+    pub name: Vec<u8>,
+    pub description: Vec<u8>,
+    pub admin_pubkeys: Vec<Vec<u8>>,
+    pub relays: Vec<Vec<u8>>,
+}
+
+/// This is an MLS Group Context extension used to store the group's name,
+/// description, ID, and admin identities.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NostrGroupDataExtension {
+    /// Nostr Group ID
+    pub nostr_group_id: [u8; 32],
```

Comment:
In the `nostr-mls-storage` crate the `nostr_group_id` is a string. Can we change it there to be a `[u8; 32]` too? I can do it in another PR